pkg/provision/workspace/routing.go

```diff
@@ -153,6 +154,7 @@ func getSpecRouting(
 	if val, ok := workspace.Annotations[constants.DevWorkspaceRestrictedAccessAnnotation]; ok {
 		annotations = maputils.Append(annotations, constants.DevWorkspaceRestrictedAccessAnnotation, val)
 	}
+	annotations[constants.DevWorkspaceStartedStatusAnnotation] = "true"
```

Comment:
it seems to be a place which try to write nil map.
```suggestion
	annotations = maputils.Append(annotations, constants.DevWorkspaceStartedStatusAnnotation, "true")
```